deep-learning/src/main/scala/com/microsoft/azure/synapse/ml/cntk/ImageFeaturizer.scala

```diff
@@ -48,65 +52,71 @@ class ImageFeaturizer(val uid: String) extends Transformer with HasInputCol with
 
-  val cntkModel: TransformerParam =
+  val onnxModel: TransformerParam =
     new TransformerParam(
       this,
-      "cntkModel", "The internal CNTK model used in the featurizer",
-      { t => t.isInstanceOf[CNTKModel] })
+      "onnxModel", "The internal ONNX model used in the featurizer",
+      { t => t.isInstanceOf[ONNXModel] })
 
   /** @group setParam */
-  def setCntkModel(value: CNTKModel): this.type = set(cntkModel, value)
+  def setOnnxModel(value: ONNXModel): this.type = set(onnxModel, value)
 
-  val emptyCntkModel = new CNTKModel()
+  val emptyOnnxModel = new ONNXModel()
 
   /** @group getParam */
-  def getCntkModel: CNTKModel =
-    if (isDefined(cntkModel)) $(cntkModel).asInstanceOf[CNTKModel] else emptyCntkModel
+  def getOnnxModel: ONNXModel =
+    if (isDefined(onnxModel)) $(onnxModel).asInstanceOf[ONNXModel] else emptyOnnxModel
 
   /** @group setParam */
-  def setMiniBatchSize(value: Int): this.type = set(cntkModel, getCntkModel.setMiniBatchSize(value))
+  def setMiniBatchSize(value: Int): this.type = set(onnxModel, getOnnxModel.setMiniBatchSize(value))
 
   /** @group getParam */
-  def getMiniBatchSize: Int = getCntkModel.getMiniBatchSize
+  def getMiniBatchSize: Int = getOnnxModel.getMiniBatchSize
 
-  /** @group setParam */
-  def setInputNode(value: Int): this.type = set(cntkModel, getCntkModel.setInputNodeIndex(value))
+  def setModelLocation(path: String): this.type = {
+    set(onnxModel, getOnnxModel.setModelLocation(path))
+  }
 
-  /** @group getParam */
-  def getInputNode: Int = getCntkModel.getInputNodeIndex
+  def setModel(name: String): this.type = {
+    val hub = new ONNXHub()
+    val info = hub.getModelInfo(name)
+    val bytes = hub.load(name)
+    setModel(bytes).setModelInfo(info)
+  }
 
-  def setModelLocation(path: String): this.type = {
-    set(cntkModel, getCntkModel.setModelLocation(path))
+  def setModelInfo(info: ONNXModelInfo): this.type = {
+    val input = info.metadata.ioPorts.get.inputs.head
+    setImageHeight(input.shape.head.right.get)
+      .setImageWidth(input.shape.apply(1).right.get)
+      .setImageTensorName(input.name)
+      .setFeatureTensorName(info.metadata.extraPorts.get.features.head.name)
   }
 
-  def setModel(modelSchema: ModelSchema): this.type = {
-    setLayerNames(modelSchema.layerNames)
-      .setInputNode(modelSchema.inputNode)
-      .setModelLocation(modelSchema.uri.toString)
+  def setModel(bytes: Array[Byte]): this.type = {
+    set(onnxModel, getOnnxModel.setModelPayload(bytes))
   }
 
+  /** @group getParam */
+  /** @group getParam */
+  def getModel: Array[Byte] = getOnnxModel.getModelPayload
+
+  val imageHeight: IntParam = new IntParam(
+    this, "imageHeight", "Size required by model", ParamValidators.gtEq(0))
+
   /** @group setParam */
-  def setModel(model: CNTKFunction): this.type = set(cntkModel, getCntkModel.setModel(model))
+  def setImageHeight(value: Int): this.type = set(imageHeight, value)
 
```

Comment:
Will this  auto-compute the required height based on the input node shape?

Reply:
this is your code, so you tell me the spec you wanted  :)

Reply:
right now, if you call setModel(), then yes, the width and height will be set